var/spack/repos/builtin/packages/hiredis/package.py

```diff
@@ -29,17 +24,53 @@ class Hiredis(CMakePackage, MakefilePackage):
     version("0.13.3", sha256="717e6fc8dc2819bef522deaca516de9e51b9dfa68fe393b7db5c3b6079196f78")
     version("0.13.2", sha256="b0cf73ebe039fe25ecaaa881acdda8bdc393ed997e049b04fc20865835953694")
 
+    build_system(
+        conditional("cmake", when="@1.0.0:"),
+        conditional("makefile", when="@:0.14.1"),
+        default="cmake",
+    )
+
+    variant("ssl", default=False, description="Builds with SSL enabled")
+    variant("test", default=False, description="Builds test suite")
+    variant("test_ssl", default=False, description="Builds  test suite for SSL")
+    variant("test_async", default=False, description="Builds test suite for async primitives")
 
-def darwin_fix(builder):
-    fix_darwin_install_name(builder.pkg.prefix.lib)
+    depends_on("cmake@3.18:", type="build", when="build_system=cmake")
+    depends_on("openssl@1.1:", type=("build", "link"), when="+ssl")
+    depends_on("openssl@1.1:", type=("build", "link"), when="+test_ssl")
 
 
 class MakefileBuilder(spack.build_systems.makefile.MakefileBuilder):
+    @property
+    def build_targets(self):
+        use_ssl = 1 if "+ssl" in self.spec else 0
+        run_test_async = 1 if "+test_async" in self.spec else 0
+        return [
+            "USE_SSL={0}".format(use_ssl),
+            "TEST_ASYNC={0}".format(run_test_async),
+        ]
+
     def install(self, pkg, spec, prefix):
         make("PREFIX={0}".format(prefix), "install")
+        if "+test" in self.spec or "+test_async" in self.spec or "+test_ssl" in self.spec:
+            make("PREFIX={0}".format(prefix), "test")
 
-    run_after("install", when="platform=darwin")(darwin_fix)
+    @run_after("install")
+    def darwin_fix(self):
+        if self.spec.satisfies("platform=darwin"):
+            fix_darwin_install_name(self.prefix.lib)
```

Comment:
Does cmake also need the darwin fix?  If not, let's use your version.

Reply:
I believe CMake does rpaths better